### PR TITLE
Update token module endpoint - Closes #7123

### LINKED ISSUE #7123: Update token Endpoint implementation
### Description
Implement endpoint as below

#### Req
```
{
  address: string,
}
```

#### Res
```
{
  balances: {
    tokenID: string,
    availableBalance: string,
    lockedBalances: { moduleID: number, amount: string }[],
  }[],
}
```
#### Req
```
{
  address: string,
  tokenID: string,
}
```
#### Res
```
{
  availableBalance: string,
  lockedAmount: { moduleID: number, amount: string }[],
}
```
#### Req
```
{
}
```
#### Res
```
{
  totalSupply: {
    tokenID: string,
    supply: string, // number string
  }[]
}
```
#### Req
```
{
}
```
#### Res
```
{
  tokenIDs: string[],
}
```
#### Req
```
{
}
```
#### Res
```
{
  escrowedAmounts: {
    tokenID: string,
    amount: string,
  }[]
}
```

### Acceptance Criteria
- All endpoints are unit tested

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7123

### How was it solved?

- Add token module endpoint as defined in the issue
- Add `getImmutableAPIContext` in the module endpoint context. This is to be able to access the readonly API defined in other modules. For token module, accessing the interoperability chain ID was necessary, and without access to API, it was not possible

### How was it tested?

- Add/update unit test

### CODE REVIEW

**Review by ishantiw — APPROVED** [review]

File: framework/test/unit/modules/token/endpoint.spec.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		lockedBalances: [
+			{
+				moduleID: 12,
+				amount: BigInt(100000000),
+			},
+		],
+	};
+	const defaultTotalSupply = BigInt('100000000000000');
+	const defaultEscrowAmount = BigInt('100000000000');
+	const supportedTokenIDs = ['0000000000000000', '0000000200000000'];
```
[the remark this answers is not in the document]
Comment (mehmetegemen) [reply framework/test/unit/modules/token/endpoint.spec.ts:55]: Or maybe we can create a template for tagged template literals like `_\`000_000_0_2_000_000_00\``?

File: framework/src/modules/token/endpoint.ts [hunk 2]
```diff
@@ -16,22 +16,80 @@ import { NotFoundError } from '@liskhq/lisk-chain';
...
+	}
+
+	public async getBalances(
+		context: ModuleEndpointContext,
+	): Promise<{ balances: JSONObject<UserStoreData & { tokenID: Buffer }>[] }> {
+		const errors = validator.validate(getBalancesRequestSchema, context.params);
+		if (errors.length) {
+			throw new LiskValidationError(errors);
+		}
+		const address = Buffer.from(context.params.address as string, 'hex');
```
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/src/modules/token/endpoint.ts:56]: I think we can improve the `validate` on validators. created https://github.com/LiskHQ/lisk-sdk/issues/7156 for now

File: framework/test/unit/modules/token/endpoint.spec.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	let stateStore: StateStore;
+
+	beforeEach(async () => {
+		const api = new TokenAPI(MODULE_ID_TOKEN);
+		endpoint = new TokenEndpoint(MODULE_ID_TOKEN);
+		api.init({
+			minBalances: [
+				{
+					tokenID: DEFAULT_TOKEN_ID,
+					amount: BigInt(5000000),
```
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/test/unit/modules/token/endpoint.spec.ts:67]: I am ok to use underscore, but we should separately align and make it consistent across repo.
You could create a new issue, and we can decide on one direction

File: framework/test/unit/modules/token/endpoint.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/test/unit/modules/token/endpoint.spec.ts:55]: I think it's hex value, so for me it's easier to read like this than separating

File: framework/test/unit/modules/token/endpoint.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mehmetegemen) [reply framework/test/unit/modules/token/endpoint.spec.ts:67]: Okay

**Review by mehmetegemen — APPROVED** [review]

**Pull request merged**